Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -4520,6 +4520,17 @@ static bool selectionIsInsideFixedPositionContainer(LocalFrame& frame)
     Checked<int> pageWidth = pageRects[0].width();
     Checked<int> pageHeight = pageRects[0].height();
 
+    // The thumbnail images are always a maximum of 500 x 500.
+    static constexpr float maximumPrintPreviewDimensionSize = 500.0;
+
+    // If the sizes are too large, the bitmap will not be able to be created,
+    // so scale them down.
+    float scaleFactor = maximumPrintPreviewDimensionSize / static_cast<int>(std::max(pageWidth, pageHeight));
+    if (scaleFactor < 1.0) {
+        pageWidth = static_cast<int>(std::floorf(static_cast<int>(pageWidth) * scaleFactor));
+        pageHeight = static_cast<int>(std::floorf(static_cast<int>(pageHeight) * scaleFactor));
```

Comment:
I _think_ this would be slightly cleaner as:

```
pageWidth = static_cast<int>(pageWidth.value() * scaleFactor);
pageHeight = static_cast<int>(pageHeight.value() * scaleFactor);
```